common/types.go

```diff
@@ -182,6 +182,10 @@ func (h *Hash) SetBytes(b []byte) {
 	copy(h[HashLength-len(b):], b)
 }
 
+func (h *Hash) IsEmpty() bool {
+	return *h == Hash{}
```

Comment:
should we move `Hash{}` in a constant so we don't recreate everytime a ressource?
we could also call `EmptyHash(*h)` below

Reply:
Done